vortex-io/src/object_store.rs

```diff
@@ -113,38 +117,56 @@ impl VortexReadAt for ObjectStoreReadAt {
     }
 }
 
+#[derive(Clone)]
 pub struct ObjectStoreWriter {
+    inner: Arc<Mutex<ObjectStoreWriterInner>>,
```

Comment:
All function here take a `&mut self` right? That's the guarantee against concurrent access.

Reply:
yep, good call